internal/indexer/indexer.go

```diff
@@ -197,6 +203,126 @@ func (idx *Indexer) Run() error {
 			}
 		}
 
+		// Update draft document headers, if configured.
+		if idx.UpdateDraftHeaders {
+			log.Info("refreshing draft document headers",
+				"folder_id", idx.DraftsFolderID,
+			)
+			currentTime := time.Now().UTC()
+
+			// Get drafts folder data (headers) from the database.
+			// Note: we add a "refreshHeaders:" prefix for the Google Drive ID here
+			// to not conflict with with the actual last indexed time of the folder
+			// (if we're indexing it, that is).
+			fd := models.IndexerFolder{
+				GoogleDriveID: fmt.Sprintf("refreshHeaders:%s", idx.DraftsFolderID),
+			}
+			if err := fd.Get(db); err != nil && !errors.Is(
+				err, gorm.ErrRecordNotFound) {
+				log.Error("error getting drafts headers folder indexer data",
+					"error", err,
+				)
+				os.Exit(1)
```

Comment:
Do we want to exit here? Could we retry instead?

Reply:
It's a good thought, but we're already exiting for basically all errors in the indexer (there's a TODO above to improve error handling). I'd like to address this more holistically in a future PR.